core/src/ldml/ldml_transforms.hpp

```diff
@@ -25,11 +26,19 @@
 #include "unicode/unistr.h"
 #include "unicode/regex.h"
 #include "unicode/utext.h"
+#include "unicode/normalizer2.h"
 
 namespace km {
 namespace kbp {
 namespace ldml {
 
+// macro for working with ICU calls
+#define UASSERT_SUCCESS(status) \
+  if (U_FAILURE(status)) { \
+    DebugLog("U_FAILURE(%s)", u_errorName(status)); \
+    assert(U_SUCCESS(status)); \
+  }
```

Comment:
What do we do in failure cases in release builds, where the assert won't happen? We'll get different behaviour because it will fall through and continue. Do we need to return false or something in the function?

Also might be better to do this as an inline function because it can be optimized more cleanly and avoids accidental misuse with function calls as the status parameter etc.

Reply:
changed to return false. 

Note that the call site _already_ kept the string unchanged if the assert failed.

Reply:
I also improved the transform loading code to handle assert failures as validation failures